apps/src/code-studio/components/progress/UnitOverviewTopRow.jsx

```diff
@@ -79,9 +80,14 @@ class UnitOverviewTopRow extends React.Component {
     const {
       scriptOverviewPdfUrl,
       scriptResourcesPdfUrl,
-      publishedState
+      publishedState,
+      noLessonPlans
     } = this.props;
 
+    if (noLessonPlans) {
+      return [];
+    }
+
```

Comment:
This looks like it will work, but also it looks like the logic for whether we might show this menu is getting pretty spread out:
* https://github.com/code-dot-org/code-dot-org/blob/76664db7c1cb2c4163cc95f9989ca9e044efc2c0/dashboard/lib/services/curriculum_pdfs/script_overview.rb#L11-L12
* https://github.com/code-dot-org/code-dot-org/blob/76664db7c1cb2c4163cc95f9989ca9e044efc2c0/dashboard/lib/services/curriculum_pdfs/script_overview.rb#L19-L21
* https://github.com/code-dot-org/code-dot-org/blob/fdb47d146e19b189399d9e8d82f5e3da970c60f1/dashboard/app/models/script.rb#L2066-L2067

I'm not sure it is worth reimplementing at this time, but perhaps a good convention for the future would be to try to move the logic as far upstream as possible? For example, in this case, maybe we shouldn't even be generating these PDFs if there are no lesson plans.

Reply:
Thats a great point Dave. I will try to do a pass to refactor this.